python/ray/tune/impl/tuner_internal.py

```diff
@@ -218,6 +233,18 @@ def _restore_from_path_or_uri(
             tuner = pickle.load(fp)
             self.__dict__.update(tuner.__dict__)
 
+        # Re-wrap the trainable with newly-specified parameters, if provided
+        # If not, throw an error when trying to fit
+        if hasattr(trainable, "_unwrapped_trainable_and_params"):
+            (
+                unwrapped_trainable,
+                attached_params,
+            ) = trainable._unwrapped_trainable_and_params
+            if not parameters or set(parameters.keys()) != set(attached_params):
```

Comment:
I think we can just do
```suggestion
            if not parameters or set(parameters) != set(attached_params):
```
(without `.keys()`)